src/callbacks/callback_print.cpp

```diff
@@ -192,24 +194,36 @@ void lbann_callback_print::report_results(model *m) {
         std::vector<int> num_samples_list(comm->get_num_trainers());
         comm->intertrainer_gather(score, score_list);
         comm->intertrainer_gather(score_samples, num_samples_list);
-        for (int i = 0; i < num_models; ++i) {
-          std::cout << m->get_name() << " (instance " << i <<  ") " << mode_string << " "
-                    << met->name() << " : "
-                    << score_list[i] << met->get_unit()
-                    << std::endl;
+        if(!m_print_global_stat_only) {
+          for (int i = 0; i < num_trainers; ++i) {
+            std::cout << m->get_name() << " (instance " << i <<  ") " << mode_string << " "
+                      << met->name() << " : "
+                      << score_list[i] << met->get_unit()
+                      << std::endl;
+          }
         }
-        if (num_models > 1) {
+        if (num_trainers > 1) {
+          const EvalType min_score = *std::min_element(score_list.begin(), score_list.end());
           const EvalType avg_score = (std::inner_product(num_samples_list.begin(),
                                                          num_samples_list.end(),
                                                          score_list.begin(),
                                                          EvalType(0))
                                       / std::accumulate(num_samples_list.begin(),
                                                         num_samples_list.end(),
                                                         0));
-          std::cout << m->get_name() << " (global) "  << mode_string << " "
+          const EvalType max_score = *std::max_element(score_list.begin(), score_list.end());
+          std::cout << m->get_name() << " (global min) "  << mode_string << " "
+                    << met->name() << " : "
+                    << min_score << met->get_unit()
+                    << std::endl;
+          std::cout << m->get_name() << " (global average) "  << mode_string << " "
                     << met->name() << " : "
                     << avg_score << met->get_unit()
                     << std::endl;
+          std::cout << m->get_name() << " (global max) "  << mode_string << " "
+                    << met->name() << " : "
+                    << max_score << met->get_unit()
+                    << std::endl;
```

Comment:
Can we change the order to print mean, then min, then max? It's more intuitive to have the min and max next to each other.

Reply:
Done.